In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [ ]:
data = pd.read_csv("Dataset of Diabetes .csv")
df = data.copy()
df.head()

ID  No_Pation Gender  AGE  Urea  Cr  HbA1c  Chol   TG  HDL  LDL  VLDL  \
0  502      17975      F   50   4.7  46    4.9   4.2  0.9  2.4  1.4   0.5   
1  735      34221      M   26   4.5  62    4.9   3.7  1.4  1.1  2.1   0.6   
2  420      47975      F   50   4.7  46    4.9   4.2  0.9  2.4  1.4   0.5   
3  680      87656      F   50   4.7  46    4.9   4.2  0.9  2.4  1.4   0.5   
4  504      34223      M   33   7.1  46    4.9   4.9  1.0  0.8  2.0   0.4   

    BMI CLASS  
0  24.0     N  
1  23.0     N  
2  24.0     N  
3  24.0     N  
4  21.0     N

In [ ]:
df.shape

(1000, 14)

* Removing noises in the target features

In [ ]:
df["CLASS"].unique()

array(['N', 'N ', 'P', 'Y', 'Y '], dtype=object)

In [ ]:
# I am removing the features with class label is not "Y" and "N"
df = df[df["CLASS"].isin(["Y","N"])]
df.shape

(942, 14)

### Removing the constant features
* ID
* No_Pation

In [ ]:
df.drop(columns=["ID","No_Pation"],inplace=True)
df.head(2)

Gender  AGE  Urea  Cr  HbA1c  Chol   TG  HDL  LDL  VLDL   BMI CLASS
0      F   50   4.7  46    4.9   4.2  0.9  2.4  1.4   0.5  24.0     N
1      M   26   4.5  62    4.9   3.7  1.4  1.1  2.1   0.6  23.0     N

In [ ]:
df["Gender"].value_counts()

M    524
F    417
f      1
Name: Gender, dtype: int64

In [ ]:
df["Gender"][df["Gender"] == "f"] = "F"

<ipython-input-161-9d5b6730a494>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["Gender"][df["Gender"] == "f"] = "F"


In [ ]:
df["Gender"].value_counts()

M    524
F    418
Name: Gender, dtype: int64

In [ ]:
df["CLASS"].value_counts()

Y    840
N    102
Name: CLASS, dtype: int64

* All noises in the catagorical features are now cleared

In [ ]:
from sklearn.preprocessing import LabelEncoder
encoder = LabelEncoder()

In [ ]:
features_to_encode = ["CLASS","Gender"]
for features in features_to_encode:
  df[features] = encoder.fit_transform(df[features])
df.head()

Gender  AGE  Urea  Cr  HbA1c  Chol   TG  HDL  LDL  VLDL   BMI  CLASS
0       0   50   4.7  46    4.9   4.2  0.9  2.4  1.4   0.5  24.0      0
1       1   26   4.5  62    4.9   3.7  1.4  1.1  2.1   0.6  23.0      0
2       0   50   4.7  46    4.9   4.2  0.9  2.4  1.4   0.5  24.0      0
3       0   50   4.7  46    4.9   4.2  0.9  2.4  1.4   0.5  24.0      0
4       1   33   7.1  46    4.9   4.9  1.0  0.8  2.0   0.4  21.0      0

* All features are encoded now its time for splitting the dataset in training and testing dataset

In [ ]:
X = df.drop(columns=["CLASS"])
Y = df["CLASS"]

In [ ]:
from sklearn.model_selection import train_test_split
X_train,X_test,Y_train,Y_test = train_test_split(X,Y,train_size=0.8,random_state=42,shuffle=True)

In [ ]:
X_test.shape,X_train.shape

((189, 11), (753, 11))

In [ ]:
Y_train.shape,Y_test.shape

((753,), (189,))

In [ ]:
# Using Random forest classfier since the dataset is imbalaced
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import confusion_matrix,accuracy_score

# training model
model = RandomForestClassifier()
model.fit(X_train,Y_train)
y_pred = model.predict(X_test)

# Accuracy score
acuracy_score = accuracy_score(y_pred,Y_test)
print("Accuracy scaore: ",acuracy_score)

# confusion matrics
matrix = confusion_matrix(Y_test,y_pred)
print(matrix)


Accuracy scaore:  0.9894179894179894
[[ 18   1]
 [  1 169]]


* Without any transformation, or any binning
* Accuracy scaore:  0.9894179894179894


In [ ]:
# finished